armi/reactor/converters/blockConverters.py

```diff
@@ -602,10 +603,11 @@ def _classifyComponents(self):
 
     def _buildFirstRing(self, pinComponents):
         """Add first ring of components to new block."""
-        newComps = copy.deepcopy(pinComponents)
-        for c in newComps:
+        for oldC in pinComponents:
+            c = copy.deepcopy(oldC)
```

Comment:
I also considered using `composites.Composite.setName()` instead of `component.setName()` here to avoid clobbering the flags. It would be a one-line change instead of 3, but then we have to import `composites` here so maybe this is better.